pyro/infer/mcmc/hmc.py

```diff
@@ -307,10 +321,13 @@ def initial_trace(self):
         trace = poutine.trace(self.model).get_trace(*self._args, **self._kwargs)
         for i in range(self._max_tries_initial_trace):
             trace_log_prob_sum = self._compute_trace_log_prob(trace)
+            z = {name: node["value"].detach() for name, node in self._iter_latent_nodes(trace)}
+            #print("in configure", z)
+            #print(trace_log_prob_sum)
             if not torch_isnan(trace_log_prob_sum) and not torch_isinf(trace_log_prob_sum):
                 self._initial_trace = trace
                 return trace
-            trace = poutine.trace(self.model).get_trace(*self._args, **self._kwargs)
+            trace = poutine.trace(self.model).get_trace(self._args, self._kwargs)
```

Comment:
This seems to be a regression?

Reply:
Sorry, I'll go for a pass to clean these diffs. Currently, NUTS is fast in double precision for your example, but slow in single precision. I'm investigating it.